src/backend/executor/execMain.c

```diff
@@ -2264,11 +2264,13 @@ InitPlan(QueryDesc *queryDesc, int eflags)
 			/*
 			 * A subplan will never need to do BACKWARD scan nor MARK/RESTORE.
 			 *
-			 * GPDB: We always set the REWIND flag, to delay eagerfree.
+			 * GPDB: We always set the REWIND flag, except InitPlans,
+			 * to delay eagerfree.
 			 */
 			sp_eflags = eflags
 				& (EXEC_FLAG_EXPLAIN_ONLY | EXEC_FLAG_WITH_NO_DATA);
-			sp_eflags |= EXEC_FLAG_REWIND;
+			if (bms_is_member(subplan_idx + 1, plannedstmt->rewindPlanIDs))
```

Comment:
If ORCA does not generate init plans, how about such
```suggestion
			if (plannedstmt->planGen == PLANGEN_OPTIMIZER || bms_is_member(subplan_idx + 1, plannedstmt->rewindPlanIDs))
```
instead all code changes in ORCA?

Reply:
Done

Reply:
Can you explain how absence of init-plan generation in ORCA is connected with adding one more kludge with planner-dependent logic in the executor? What problems with initial solution, that modifying ORCA?

Reply:
>What problems with initial solution, that modifying ORCA?

Do we need to multiply entities beyond necessity?

Reply:
If it makes code more straightforward, I think we can. We have `rewindPlanIDs` in the Planned Statement structure. Do we really want to care about the kind of planner that produce this plan to understand that we can rely on the value stored in this field? I could understand you if it requires many efforts. The less the executor knows about the specific of the exact planner, the better.

Reply:
Fair point. I'll return ORCA changes